protocol/0083-RFPR-on_chain_referral_program.md

```diff
@@ -329,20 +359,44 @@ Given:
         {
             "minimum_running_notional_taker_volume": 10000,
             "minimum_epochs": 0,
-            "referral_reward_factor": 0.001,
-            "referral_discount_factor": 0.001,
+            "referral_reward_factor": {
+                "maker_volume_discount_factor": 0.001,
+                "infrastructure_volume_discount_factor": 0.001,
+                "lp_volume_discount_factor": 0.001
+            },
+            "referral_discount_factor": {
+                "maker_volume_discount_factor": 0.001,
+                "infrastructure_volume_discount_factor": 0.001,
+                "lp_volume_discount_factor": 0.001
+            },
         },
         {
             "minimum_running_notional_taker_volume": 20000,
             "minimum_epochs": 7,
-            "referral_reward_factor": 0.005,
-            "referral_discount_factor": 0.005,
+            "referral_reward_factor": {
+                "maker_volume_discount_factor": 0.005,
+                "infrastructure_volume_discount_factor": 0.005,
+                "lp_volume_discount_factor": 0.005
+            },
+            "referral_discount_factor": {
+                "maker_volume_discount_factor": 0.005,
+                "infrastructure_volume_discount_factor": 0.005,
+                "lp_volume_discount_factor": 0.005
+            },
         },
         {
             "minimum_running_notional_taker_volume": 30000,
             "minimum_epochs": 31,
-            "referral_reward_factor": 0.010,
-            "referral_discount_factor": 0.010,
+            "referral_reward_factor": {
+                "maker_volume_discount_factor": 0.01,
+                "infrastructure_volume_discount_factor": 0.01,
+                "lp_volume_discount_factor": 0. 01
```

Comment:
Ditto here, `reward` not `discount` and match `0029`.

And should be `referral` not `volume`.